File: backend/app/repo/models/acceptor.py
from sqlalchemy import Column, ForeignKey, Unicode, Boolean

from app.repo.models.base import Model


class Acceptor(Model):
    name = Column(Unicode)
    system_id = Column(ForeignKey('system.id'))
    user_id = Column(ForeignKey('user.id'))
    account = Column(Unicode)
    is_disabled = Column(Boolean, default=False)
    is_system = Column(Boolean, default=False)

File: backend_go/src/repo/models/acceptor.go
package models

type Acceptor struct {
	Id         int    `json:"id"`
	IsDeleted  bool   `json:"is_deleted"`
	IsDisables bool   `json:"is_disabled"`
	IsSystem   bool   `json:"is_system"`
	Name       string `json:"name"`
	UserId     int    `json:"user_id"`
	SystemId   int    `json:"system_id"`
	Account    string `json:"account"`
}

func (n *Acceptor) GetEmptyItem() (any, any) {
	inst := Acceptor{}
	return inst, &inst
}

func (n *Acceptor) MapRecords(prepared []interface{}) []Acceptor {
	var result = make([]Acceptor, 0)
	for _, element := range prepared {
		result = append(result, element.(Acceptor))
	}

	return result
}
